### PR TITLE
:sparkles: Removed array concept from emit and listen

### PR DESCRIPTION
**Pull request by Rayyan98**

### CODE REVIEW

File: src/emitters/emitter.sqns.ts [hunk 1]
```diff
@@ -385,13 +386,13 @@ export class SqnsEmitter implements IEmitter {
...
       };
       let response = false;
-      const modifiedArgs = (await this.options.hooks?.beforeEmit?.(eventName, args)) || args;
+      const modifiedArgs = (await this.options.hooks?.beforeEmit?.(eventName, payload)) || payload;
       if (topic.exchangeType === ExchangeType.Queue) {
-        response = await this.emitToQueue(topic, options, ...modifiedArgs);
+        response = await this.emitToQueue(topic, options, modifiedArgs);
       } else {
-        response = await this.emitToTopic(topic, options, ...args);
+        response = await this.emitToTopic(topic, options, payload);
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:393]:
```suggestion
        response = await this.emitToTopic(topic, options, modifiedArgs);
```

File: src/emitters/emitter.sqns.ts [hunk 2]
```diff
@@ -385,13 +386,13 @@ export class SqnsEmitter implements IEmitter {
...
       }
-      await this.options.hooks?.afterEmit?.(eventName, args);
+      await this.options.hooks?.afterEmit?.(eventName, payload);
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:395]:
```suggestion
      await this.options.hooks?.afterEmit?.(eventName, modifiedArgs);
```

File: src/emitters/emitter.sqns.ts [hunk 3]
```diff
@@ -400,10 +401,10 @@ export class SqnsEmitter implements IEmitter {
       this.logFailedEvent({
         failureType: FailedEventCategory.MessageProducingFailed,
         topic: eventName,
-        event: args,
+        event: payload,
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:404]: modifiedArgs

File: src/emitters/emitter.sqns.ts [hunk 4]
```diff
@@ -716,6 +717,13 @@ export class SqnsEmitter implements IEmitter {
       });
       throw new Error(`Failed to parse message`);
     }
+
+    const payloadStructureVersion = message.messageAttributes?.PayloadVersion.StringValue;
```
Comment (hammad20120) [inline src/emitters/emitter.sqns.ts:721]: tested this locally, need to update parseDataFromMessage to parse messageAttributes as well.
for SQS it is in `receivedMessage.messageAttributes` 
for SNS it is in  `JSON.parse(receivedMessage.body).MessageAttributes`

Comment (hammad20120) [reply src/emitters/emitter.sqns.ts:721]: Also for SNS the structure seems to be `PayloadVersion.Value`
For SQS the structure seems to be `PayloadVersion.stringValue`

File: src/emitters/emitter.sqns.ts (hunk 1 above)
Comment (Rayyan98) [reply src/emitters/emitter.sqns.ts]: Done

File: src/emitters/emitter.sqns.ts (hunk 2 above)
Comment (Rayyan98) [reply src/emitters/emitter.sqns.ts]: Done

File: src/emitters/emitter.sqns.ts (hunk 3 above)
Comment (Rayyan98) [reply src/emitters/emitter.sqns.ts]: Done

File: src/emitters/emitter.sqns.ts (hunk 4 above)
Comment (MGB247) [reply src/emitters/emitter.sqns.ts]: resolved
please verify

**Pull request closed without merging**